src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -38,27 +39,34 @@ public class CosmosDbLeaseStore : ILeaseStore
         public async Task InitialiseAsync()
         {
             await _documentClient.CreateDatabaseIfNotExistsAsync(new Database {Id = _options.Value.Database})
-                .ConfigureAwait(false);
-
+                    .ConfigureAwait(false);
+            
             DocumentCollection collectionDefinition = new DocumentCollection
             {
                 Id = _options.Value.Collection
             };
 
             collectionDefinition.PartitionKey.Paths.Add("/leaseType");
 
+            var leaseConstraint = new UniqueKey
+            {
+                Paths = new Collection<string> { "/leaseType" }
+            };
+
+            collectionDefinition.UniqueKeyPolicy.UniqueKeys.Add(leaseConstraint);
```

Comment:
Are we going to have a single collection for election requests and winner lease or will they be separated? @faprovitolo ?

Reply:
@nikhilgirraj what will happen if this code is executed on an existing collection? Will it fail or silently ignore the constraint?

Reply:
different collections for both purposes - a new collection for all election requests is part of the next PR

Reply:
It silently ignores the fact that the new request also specifies a Unique Key constraint.

Reply:
yes there will be a separate collection for leaseRequests having a relatively short DefaultTimeToLive, e.g. 30 seconds